src/config.cpp

```diff
@@ -59,6 +67,106 @@ void parse_uint32(void* value_ptr, const char* txt) {
     cfgval->set = true;
 }
 
+// Helper function to trim whitespace and remove quotations
+std::string trim(const std::string& str) {
+    size_t first = str.find_first_not_of(" \t\n\""); // Also include quotes
+    size_t last = str.find_last_not_of(" \t\n\"");
+
+    // Check for valid range
+    if (first == std::string::npos || last == std::string::npos) {
+        return str; // Empty or only whitespace/quotes
+    }
+
+    // Extract the trimmed and unquoted substring
+    std::string trimmed = str.substr(first, (last - first + 1));
+
+    // Check if the trimmed string still starts and ends with quotes
+    if (trimmed.front() == '"' && trimmed.back() == '"') {
+        return trimmed.substr(1, trimmed.size() - 2); // Remove the quotes
+    } else {
+        return trimmed;
+    }
+}
+
+void read_config_file(std::unordered_map<std::string, std::string>& umap,
+                      std::ifstream& config_stream) {
+
+    std::string line;
+    while (std::getline(config_stream, line)) {
+        // Ignore comments and empty lines
+        if (line.empty() || line[0] == '#') {
+            continue;
+        }
+        line = trim(line);
+        // Check for section headers
+        if (line[0] == '[' && line[line.size() - 1] == ']') {
+            continue;
+        } else {
+            // Parse key-value pairs
+            size_t pos = line.find('=');
+            if (pos != std::string::npos) {
+                std::string key = trim(line.substr(0, pos));
+                std::string value = trim(line.substr(pos + 1));
+                // Store values (if any)
+                if (value != "") {
+                    umap[key] = value;
+                }
+            }
+        }
+    }
+    config_stream.close();
+}
+
+void parse_config_file() {
+    std::unordered_map<std::string, std::string> file_config_values;
+    std::string conf_file = "clvk.conf";
+    std::ifstream config_stream;
+
+    std::vector<std::string> config_file_paths;
+    config_file_paths.push_back("/etc/clvk.conf");
+    config_file_paths.push_back("~/.config/clvk.conf");
```

Comment:
`~` won't work with windows. I supposed there is something is `std::filesystem` to have the a portable way to express the user directory.

Reply:
leaving this as it is since theres nothing in ```std::filesystem```